android/src/main/java/com/plaid/PlaidModule.kt

```diff
@@ -75,24 +70,17 @@ class PlaidModule internal constructor(reactContext: ReactApplicationContext) :
     reactApplicationContext.removeActivityEventListener(this)
   }
 
-  fun getLinkTokenConfiguration(data: String): LinkTokenConfiguration? {
-    val obj = JSONObject(data)
+  private fun getLinkTokenConfiguration(obj: JSONObject, token: String?): LinkTokenConfiguration? {
     val extrasMap = mutableMapOf<String, String>()
     maybePopulateExtrasMap(obj, extrasMap)
 
     val logLevel = LinkLogLevel.ASSERT
 
-    val publicKey = maybeGetStringField(obj, PUBLIC_KEY)
-
-    // If we're initializing with a Link token, we will not use or
-    // accept many of the client-side configs.
-    val token = maybeGetStringField(obj, TOKEN)
-
-    if (publicKey == null && token == null) {
-      throw IllegalStateException("Token must be part of configuration.")
+    if (token == null) {
+      return null
     }
 
-    if (token == null) {
+    if (!token.startsWith(LINK_TOKEN_PREFIX)) {
```

Comment:
HAHA update mode needs to work with LinkToken

Reply:
Clarification: I believe public-sandbox-uuid needs to work without a public key?